src/renderer/routes/flow_chart/views/ControlBar/FlowControlButtons.tsx

```diff
@@ -1,102 +1,87 @@
-    serverStatus === IServerStatus.OFFLINE;
-  const cancelFC = () => {
-    if (project.rfInstance && project.rfInstance.nodes.length > 0) {
-      cancelFlowChartRun(project.rfInstance, socketId);
-    } else {
-      alert("There is no running job on server.");
-    }
-  };
-  const onRun = async (nodes: Node<ElementsData>[], edges: Edge[]) => {
-    if (project.rfInstance && nodes.length > 0) {
-      if (_.some(nodes, (n) => n.data.invalid)) {
-        toast.error(
-          "Unknown blocks found, these must be removed before attempting to run the flow chart.",
-        );
-        return;
-      }
+    serverStatus === ServerStatus.CONNECTING ||
+    serverStatus === ServerStatus.OFFLINE;
 
-      // Only update the react flow instance when required.
-      const updatedRfInstance = {
-        ...project.rfInstance,
-        nodes,
-        edges,
-      };
+  const onRun = async () => {
+    if (nodes.length === 0) {
+      toast.info(
+        "There is no program to send to server. \n Please add at least one node first.",
+      );
+      return;
+    }
+    if (_.some(nodes, (n) => n.data.invalid)) {
+      toast.error(
+        "Unknown blocks found, these must be removed before attempting to run the flow chart.",
+      );
+      return;
+    }
 
-      setProject({
-        ...project,
-        rfInstance: updatedRfInstance,
-      });
+    sendProgramToMix(nodes, true, false);
+    wipeBlockResults();
 
-      sendProgramToMix(project.rfInstance.nodes, true, false);
-      // setProgramResults([]);
-      saveAndRunFlowChartInServer({
-        rfInstance: updatedRfInstance,
-        jobId: socketId,
-        settings: settings.filter((setting) => setting.group === "backend"),
+    const res = await runFlowchart({
+      nodes,
+      edges,
+      jobId: socketId,
+      settings: backendSettings,
+    });
+    if (res.isErr()) {
+      toast.error("Failed to run flowchart", {
+        description: res.error.message,
       });
-      setNodeParamChanged(false);
-    } else {
-      alert(
-        "There is no program to send to server. \n Please add at least one node first.",
-      );
     }
-  };
-  const { nodes, edges } = useFlowChartGraph();
 
-  const handleClick = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
-    e.preventDefault();
-    onRun(nodes, edges);
+    resetNodeParamChanged();
   };
 
-  useKeyboardShortcut("ctrl", "p", () => onRun(nodes, edges));
-  useKeyboardShortcut("meta", "p", () => onRun(nodes, edges));
+  useKeyboardShortcut("ctrl", "p", onRun);
+  useKeyboardShortcut("meta", "p", onRun);
 
   return (
     <>
-      {playBtnDisabled || serverStatus === IServerStatus.STANDBY ? (
+      {playBtnDisabled || serverStatus === ServerStatus.STANDBY ? (
         <Button
           data-cy="btn-play"
           data-testid="btn-play"
           variant="dotted"
           id="btn-play"
-          onClick={handleClick}
+          onClick={(e) => {
+            e.preventDefault();
```

Comment:
Curious to know why we need to call `e.preventDefault()` here? Any reason for not passing `onRun` directly `onClick={onRun}` ?

Reply:
Not sure, it was just like that before so i kept it that way, removing it doesn't seem to change anything though